### PR TITLE
Add vignette (again)

### LINKED ISSUE #35: Add option to scale data
See #31.

### LINKED ISSUE #38: Bayes Logistic regression in Stan
We can sidestep the BayesLogit and PolyaGamma issue by doing Bayesian Logistic regression on Stan.

### LINKED ISSUE #40: Refactor into a function that takes a Stan model as input
Simon:

> Another alternative is to get a Stan model as an input and use that. This function would call the first (in #39) one for NPL correction.

### LINKED ISSUE #45: Separate code into vignette and package

### PR DESCRIPTION
**Pull request by miguelmorin**
This was a big task and the commits up to 6191c73 close #45.

In the process, I refactored code that closes other issues. Commit 94ee515 closes #35. Commit 0999b5f closes #40.

As this pull request was open and Travis takes some time to run, and I was working intensely on it to wait for a merge or to deal with rebasing, I did the rest of the work on the package here too. For example, commit 2fcc697 closes #38.

### CODE REVIEW

**Comment by miguelmorin** [comment]
Commit 94ee515 closes #35 .

**Comment by miguelmorin** [comment]
Commit 2fcc697 closes #38.

File: R/PosteriorBootstrap.R [hunk 1]
```diff
@@ -0,0 +1,486 @@
...
+  # German statlog dataset outcomes are (1, 2), so
+  # subtract 1 here to make them (0, 1)
+  y <- y - 1
+  stopifnot(all(y %in% c(0, 1)))
+
+  # Standardise features to have mean 0 and variance 1
+  if (scale) {
+    x <- scale(x)
+  }
+  if (add_intercept) {
```
Comment (jemrobinson) [inline R/PosteriorBootstrap.R]: Why is this named "add_intercept"?

File: R/PosteriorBootstrap.R [hunk 2]
```diff
@@ -0,0 +1,486 @@
...
+  # The algorithm is statistically guaranteed to finish because the last value
+  # of the `stick_remaining` array is a product of numbers between 0 and 1, so
+  # increasing the number of values in that product drives it to zero (even if
+  # we draw new values each time).
+
+  num_stick_breaks <- min_stick_breaks
+  while (TRUE) {
+    # Draw `num_stick_breaks` independent numbers from the
+    # Beta(1, concentration) distribution. All numbers are between 0 and 1.
+    beta_draws <- stats::rbeta(n = num_stick_breaks,
```
Comment (jemrobinson) [inline R/PosteriorBootstrap.R]: Is this deterministic - so we're getting the same random numbers at each iteration?

File: R/PosteriorBootstrap.R (hunk 1 above)
Comment (miguelmorin) [reply R/PosteriorBootstrap.R]: Tongue-in-cheek: it should be called "constant_term", as "intercept" is the name of the regression coefficient on that constant term.

**Pull request merged**